### PR TITLE
Cleanup filesystem libs

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Remove del. fs-extra has `remove` which does the same thing
- Remove our own mkdir, the bug it tried to fix doesn't affect fs-extra (and is long fixed otherwise)
- Remove our own rmrf, it was another copy of del
- Fixup some type errors caused by `delete` that keep showing up every time I run tests

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/shared/sam/debugger/samDebugConfigProvider.test.ts [hunk 1]
```diff
@@ -87,10 +87,11 @@ function assertEqualLaunchConfigs(actual: SamLaunchRequestArgs, expected: SamLau
     // Remove noisy properties before doing a deep-compare.
     for (const o of [actual, expected]) {
         delete o.manifestPath
-        delete o.documentUri
-        delete o.templatePath
-        delete o.workspaceFolder
-        delete o.codeRoot
+        // the next for are as any to prevent type issues (cannot delete required fields)
```
Comment (justinmk3) [inline src/test/shared/sam/debugger/samDebugConfigProvider.test.ts:90]: looks like a typo or something. Could just omit the comment though, the `as any` is reasonably self-evident for typescript programmers.

**Review by justinmk3 — APPROVED** [review]
Wonderful, thank you :)

File: src/test/shared/sam/debugger/samDebugConfigProvider.test.ts (hunk 1 above)
Comment (hunterwerlla) [reply src/test/shared/sam/debugger/samDebugConfigProvider.test.ts]: Reasonable, changed

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**